presto-native-sidecar-plugin/src/test/java/com/facebook/presto/sidecar/TestNativeSidecarPluginSystemPropertyProvider.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sidecar;
+
+import com.facebook.presto.nativeworker.PrestoNativeQueryRunnerUtils;
+import com.facebook.presto.testing.MaterializedResult;
+import com.facebook.presto.testing.MaterializedRow;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import org.intellij.lang.annotations.Language;
+import org.testng.annotations.Test;
+
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static com.facebook.presto.sidecar.NativeSidecarPluginQueryRunnerUtils.setupNativeSidecarPlugin;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertFalse;
+
+@Test(singleThreaded = true)
+public class TestNativeSidecarPluginSystemPropertyProvider
+        extends AbstractTestQueryFramework
+{
+    private static final String REGEX_SESSION_NAMESPACE = "Native Execution only.*";
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        DistributedQueryRunner queryRunner = (DistributedQueryRunner) PrestoNativeQueryRunnerUtils.createQueryRunner(false, true);
+        setupNativeSidecarPlugin(queryRunner);
+        return queryRunner;
+    }
+
+    @Override
+    protected QueryRunner createExpectedQueryRunner()
+            throws Exception
+    {
+        return PrestoNativeQueryRunnerUtils.createJavaQueryRunner();
+    }
+
+    private List<MaterializedRow> excludeSystemSessionProperties(List<MaterializedRow> inputRows)
+    {
+        return inputRows.stream()
+                .filter(row -> Pattern.matches(REGEX_SESSION_NAMESPACE, row.getFields().get(4).toString()))
+                .collect(Collectors.toList());
+    }
+
+    @Test
+    public void testShowSession()
+    {
+        @Language("SQL") String sql = "SHOW SESSION";
+        MaterializedResult actualResult = computeActual(sql);
+        List<MaterializedRow> actualRows = actualResult.getMaterializedRows();
+        List<MaterializedRow> filteredRows = excludeSystemSessionProperties(actualRows);
+        assertFalse(filteredRows.isEmpty());
+    }
+
+    @Test
+    public void testJavaWorkerProperties()
```

Comment:
why does this fail.  I thought the Java properties are registered even when using native worker.  Can you add a test (maybe i missed it?) of reading java properties on a native worker setup and native properties in a java worker set up?

Reply:
Added the test cases :
For native properties in a java worker: https://github.com/prestodb/presto/blob/4eb5cf4a6017ca11ee531164114f1f20f35a2422/presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java#L

For java properties in a native worker setup:
https://github.com/prestodb/presto/blob/4eb5cf4a6017ca11ee531164114f1f20f35a2422/presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeGeneralQueries.java#L1178